apps/workplace/src/app/book/new-desk-flow/new-desk-form-details.component.ts

```diff
@@ -0,0 +1,306 @@
+                        </button>
+                    </div>
+                </div>
+                <button
+                    mat-button
+                    add-space
+                    class="w-full inverse mt-2"
+                    (click)="selectDesk()"
+                >
+                    <div class="flex items-center justify-center space-x-2">
+                        <app-icon>search</app-icon>
+                        <span>Find Desk</span>
+                    </div>
+                </button>
+            </section>
+            <section class="p-2">
+                <h3 class="space-x-2 flex items-center mb-4">
+                    <div
+                        class="border border-black rounded-full h-6 w-6 flex items-center justify-center"
+                    >
+                        3
+                    </div>
+                    <div class="text-xl">Request Asset</div>
+                </h3>
+                <asset-list-field formControlName="assets"></asset-list-field>
+            </section>
+        </div>
+    `,
+})
+export class NewDeskFormDetailsComponent {
+    @Input() public form: FormGroup;
+    @Output() public find = new EventEmitter<void>();
+    /** List of available buildings to select */
+    public readonly buildings = this._org.building_list;
+    /** List of available levels for the selected building */
+    public readonly levels = this._org.active_levels;
+    /** List of set options for desk booking */
+    public readonly options = this._state.options;
+    /** List of set options for desk booking */
+    public readonly features = this._state.features;
+
+    /** Selected desk for booking */
+    public selectedDesk: Desk;
+
+    public readonly recurrence_options = ['daily', 'weekly', 'monthly'];
+
+    public get building() {
+        return this._org.building;
+    }
+    public set building(bld) {
+        this._org.building = bld;
+    }
+
+    public readonly setOptions = (o) => this._state.setOptions(o);
+
+    public readonly setFeature = (f, e) => this._state.setFeature(f, e);
+
+    public get can_book_for_others() {
+        return this._settings.get('app.desks.can_book_for_others');
+    }
+    public get can_recurr() {
+        return this._settings.get('app.desks.recurrence_allowed');
+    }
+
+    public get allow_groups() {
+        return this._settings.get('app.desks.allow_groups');
+    }
+
+    public get needs_reason() {
+        return this._settings.get('app.desks.needs_reason') === true;
+    }
+
+    public get allow_time_changes() {
+        return !!this._settings.get('app.desks.allow_time_changes');
+    }
+
+    public get allow_all_day() {
+        return (
+            this.allow_time_changes &&
+            !!this._settings.get('app.desks.allow_all_day')
+        );
+    }
+
+    public get book_until() {
+        return endOfDay(
+            addDays(
+                Date.now(),
+                this._settings.get('app.desks.available_period') || 90
```

Comment:
Hmm... that settings is for less than day booking. e.g. A 90 minute booking.
If this is for multi-day booking add a new setting key for it.

Reply:
Removing as it is unused